OSSM 9894_DNS capture (#110676)
openshift/openshift-docs: modules/ossm-migrating-read-me-dns-capture-configuration.adoc

// Module included in the following assemblies:
//
// * service-mesh-docs-main/migrating/checklists/ossm-migrating-read-me.adoc

:_mod-docs-content-type: CONCEPT
[id="ossm-migrating-read-me-dns-capture-configuration_{context}"]
= DNS capture configuration for ServiceEntry resources

[role="_abstract"]

> If your workloads use `ServiceEntry` resources to access external services through DNS names, you must explicitly configure DNS capture in the proxy metadata settings when migrating to {SMProductName} 3.0. Without DNS capture enabled, applications encounter DNS resolution failures with errors such as `Name or service not known`.

Reviewer: ```suggestion
To maintain external service access during migration to {SMProductName} 3.0, you must enable DNS capture for ServiceEntry resources that use DNS resolution.
```

Author: Thanks! (And what a great sentence.) I wound up making very similar edits.

> {SMProduct} 3.0 does not enable DNS capture by default. This aligns with the removal of federation support from the upstream {istio} project. In contrast, {SMProduct} 2.6 enabled DNS capture by default through the `ISTIO_META_DNS_AUTO_ALLOCATE` and `ISTIO_META_DNS_CAPTURE` proxy metadata fields.

Reviewer 2: > This aligns with the removal of federation support from the upstream {istio} project

Upstream **Istio** only supports multi-cluster and, AFAIK, it has never supported federation (happy to be corrected). Federation was supported in {SMProduct} 2.6 and is dropped as part of {SMProduct} 3.0

Reviewer 3: oh. yeah I missed that `upstream` wording. It was dropped in OSSM 3.0 as @sridhargaddam says

Author: Thanks! I made the appropriate edits.
